src/snowflake/connector/log_configuration.py

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+
+from __future__ import annotations
+
+import logging
+import os
+from logging.handlers import TimedRotatingFileHandler
+
+from snowflake.connector.config_manager import CONFIG_MANAGER
+from snowflake.connector.constants import DIRS
+from snowflake.connector.secret_detector import SecretDetector
+
+LOG_FILE_NAME = "python-connector.log"
+
+
+class EasyLoggingConfigPython:
+    def __init__(self):
+        self.path = None
+        self.level = None
+        self.save_logs = None
+        self.parse_config_file()
+
+    def parse_config_file(self):
+        CONFIG_MANAGER.read_config()
+        data = CONFIG_MANAGER.conf_file_cache
+        if log := data.get("log"):
```

Comment:
My personal preference would be to do `if not log := data.get("log")` and `return`. In this way later code is not indented and feels easier to read

Reply:
I generally prefer positive tests over negative tests, especially if there's an `else` clause, because positive tests are easier to reason about.  This is a good case where the opportunity for an early exit and dedentation overrides that for readability.  Good catch. 

I also generally prefer more narrow conditional tests than false-yness (e.g. `if not`) but in this case because of the assignment operator, testing against `None` doesn't improve readability or understanding.